api/envoy/extensions/load_balancing_policies/common/v3/common.proto

```diff
@@ -42,8 +43,22 @@ message LocalityLbConfig {
     // failing service.
     bool fail_traffic_on_panic = 3;
 
-    // If set to true, Envoy will force LocalityDirect routing if a local locality exists.
-    bool force_locality_direct_routing = 4;
+    // If set to true, Envoy will always route traffic to the local zone regardless of the
+    // upstream zone structure. In Envoy's default configuration, traffic is distributed proportionally
+    // across all upstream hosts while trying to maximize local routing when possible. The approach
+    // with force_local_zone aims to be more predictable and if there are upstream hosts in the local
+    // zone, they will receive all traffic.
+    // * :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // * :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    bool force_local_zone = 4;
+
+    // Configures the minimum number of upstream hosts in the local zone required when force_local_zone
+    // is enabled. If the number of upstream hosts in the local zone is less than the specified value,
+    // Envoy will fall back to the default proportional-based distribution across localities.
+    // If not specified, the default is 1.
+    // * :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
+    // * :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
+    google.protobuf.UInt64Value force_local_zone_min_size = 5;
```

Comment:
Also, can we just use the presence or absence of this field to know if we should do local zoning, instead of the `force_local_zone` field?

Reply:
Could you please elaborate? `force_local_zone` was intended to be the first consideration and I'm adding `force_local_zone_min_size` as a protection mechanism for smaller zones but it's not required.

Reply:
Ah ok thanks for the explanation and nevermind on my comment then.